src/control/security/config.go

```diff
@@ -0,0 +1,91 @@
+package security
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+
+	"github.com/pkg/errors"
+)
+
+// FIXME: These should probably not be used directly;
+// instead, add a function which determines the correct
+// root path and prepends it -- need to make it easy
+// to test, though.
+const (
+	defaultCACert     = ".daos/daosCA.crt"
+	defaultServerCert = ".daos/daos_server.crt"
+	defaultServerKey  = ".daos/daos_server.key"
+	defaultClientCert = ".daos/client.crt"
+	defaultClientKey  = ".daos/client.key"
+	defaultServer     = "server"
+	defaultInsecure   = false
+)
+
+type TransportConfig struct {
+	AllowInsecure     bool `yaml:"allow_insecure"`
+	CertificateConfig `yaml:",inline"`
+}
+
+type CertificateConfig struct {
+	ServerName      string           `yaml:"server_name,omitempty"`
+	CARootPath      string           `yaml:"ca_cert"`
+	CertificatePath string           `yaml:"cert"`
+	PrivateKeyPath  string           `yaml:"key"`
+	TLSKeypair      *tls.Certificate `yaml:"-"`
+	CAPool          *x509.CertPool   `yaml:"-"`
+}
+
+// TODO: Add docs/test coverage for these functions
+
+func DefaultClientTransportConfig() *TransportConfig {
+	return &TransportConfig{
```

Comment:
since this is a subset, could read "tc := DefaultServerTransportConfig(); tc.CertificateConfig.ServerName = defaultServer; return tc"